### ISSUE TITLE
Generate RPM in addition to DEB

### ISSUE THREAD

**Issue by probonopd** [AppImage/appimaged#26]
Continuation of https://github.com/AppImage/appimaged/issues/6#issuecomment-437661060

**Comment by TheAssassin** [comment]
Not entirely sure what you're doing on that other branch, but it's clearly not a fork of the master branch. It still contains a submodule on AppImageKit?!

**Comment by TheAssassin** [comment]
You were almost there, it seems.

First of all, I ran `git fetch --all && git rebase origin/master` to update the branch.

Then, I noticed you didn't include the architecture fix snippet we use elsewhere. Since we build RPMs on a Debian system, we need to inform CPack about the right architecture. It doesn't seem very straightforward, but it seems that if that architecture is set to an invalid or empty value, a source package will be built.

```cmake
# allow building RPM packages on non-RPM systems
if(DEFINED ENV{ARCH})
    set(CPACK_RPM_PACKAGE_ARCHITECTURE $ENV{ARCH})
    if(CPACK_RPM_PACKAGE_ARCHITECTURE MATCHES "i686")
        set(CPACK_RPM_PACKAGE_ARCHITECTURE "i386")
    elseif(CPACK_RPM_PACKAGE_ARCHITECTURE MATCHES "amd64")
        set(CPACK_RPM_PACKAGE_ARCHITECTURE "x86_64")
    endif()
endif()
```

Then, I moved the `include(CPack)` at the end of both `cmake/cpack_{rpms,debs}.cmake` files into the main CMake file (below the two `include()`s), as CPack doesn't like to be included more than once (there's a warning in the output, we had the same issue in AppImageLauncher).

If you want to, I can force-push to your branch.

**Comment by probonopd** [comment]
> If you want to, I can force-push to your branch.

Yes, please do. Does it upload a rpm?

**Comment by TheAssassin** [comment]
> Travis CI still shows "green", even tough upload.sh was not successful.

That's the issue with Travis CI. It's always been behaving like that. ***Real*** CI servers have different states for this.

**Comment by probonopd** [comment]
Easy fix: Don't put stuff into `after_success` but put it in `script`.

**Comment by TheAssassin** [comment]
Travis should simply introduce a "yellow" state...

**Comment by probonopd** [comment]
True; wanna write a feature request?

**Comment by TheAssassin** [comment]
No, I'd rather switch to a real CI system that provides some real features...

**Comment by fusion809** [comment]
The RPM is not compatible with openSUSE Tumbleweed, due to missing `libglib2.0-0` dependency. If it is possible to make it more cross-platform that would be delightful.

**Comment by TheAssassin** [comment]
Closed by #27.
